src/api/add.js

```diff
@@ -117,11 +118,21 @@ async function addToIndex({
         }
       }
     } else {
+      const config = await GitConfigManager.get({ fs, gitdir })
+      const autocrlf = (await config.get('core.autocrlf')) || false
       const object = stats.isSymbolicLink()
         ? await fs.readlink(join(dir, currentFilepath)).then(posixifyPathBuffer)
-        : await fs.read(join(dir, currentFilepath))
+        : await fs.read(join(dir, currentFilepath), {
+            encoding: 'utf8',
+            autocrlf,
+          })
       if (object === null) throw new NotFoundError(currentFilepath)
-      const oid = await _writeObject({ fs, gitdir, type: 'blob', object })
+      const oid = await _writeObject({
+        fs,
+        gitdir,
+        type: 'blob',
+        object: Buffer.from(object),
```

Comment:
We wanted to get rid of Buffer, please don't add yet another use of it if possible.